src/Analyser/MutatingScope.php

```diff
@@ -4828,9 +4826,7 @@ public function processClosureScope(
 				$prevVariableType = $prevScope->getVariableType($variableName);
 				if (!$variableType->equals($prevVariableType)) {
 					$variableType = TypeCombinator::union($variableType, $prevVariableType);
-					if ($expr->getAttribute(ImmediatelyInvokedClosureVisitor::ATTRIBUTE_NAME) !== true) {
-						$variableType = self::generalizeType($variableType, $prevVariableType, 0);
-					}
+					$variableType = self::generalizeType($variableType, $prevVariableType, 0);
```

Comment:
in this file we effectively revert https://github.com/phpstan/phpstan-src/commit/8fbcf5bdc6250cb4da57ad5945f371bb0c65f931